plugin/core/windows.py

```diff
@@ -412,9 +412,29 @@ def _create_logger(self, config_name: str) -> Logger:
             return router_logger
 
     def handle_message_request(self, session: Session, params: Any, request_id: Any) -> None:
-        view = self._window.active_view()
-        if view:
-            MessageRequestHandler(view, session, request_id, params, session.config.name).show()
+        icons = {
+            1: '❗',
+            2: '⚠️',
+            3: 'ℹ️',
+            4: '📝'
+        }
+        actions = params.get("actions", [])
+        titles = list(action.get("title") for action in actions)
+
+        def send_user_choice(index: int) -> None:
+            # when noop; nothing was selected e.g. the user pressed escape
+            result = {}
```

Comment:
Shouldn't this be `None`, like in the former implementation? Specs say
> result: the selected `MessageActionItem` | `null` if none got selected.

Reply:
`None` was't serialized as `null`, this client sending `None` which was causing errors on the server.
`{}` translated to `null` with Metals at least.
Maybe Response should be updated then here: https://github.com/sublimelsp/LSP/blob/main/plugin/core/protocol.py#L393

Reply:
> this client sending None which was causing errors on the server.

That's not possible since `None` doesn't exist in JSON. Can you show communication log?

Reply:
This is the LSP log after dismissing the quick panel

```
:: <-- metals window/showMessageRequest(2): {'message': 'Select the kind of file to create', 'actions': [{'title': 'class'}, {'title': 'case-class'}, {'title': 'object'}, {'title': 'trait'}, {'title': 'package-object'}, {'title': 'worksheet'}, {'title': 'ammonite'}, {'title': 'enum'}]}
:: >>> metals 2: None
:: <~~ metals 3: {'data': 'java.util.concurrent.CompletionException: java.lang.NullPointerException\n\tat java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)\n\tat java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)\n\tat java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:661)\n\tat java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:646)\n\tat java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)\n\tat java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)\n\tat scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:29)\n\tat scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:26)\n\tat scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)\n\tat scala.concurrent.BatchingExecutor$Batch.processBatch$1(BatchingExecutor.scala:67)\n\tat scala.concurrent.BatchingExecutor$Batch.$anonfun$run$1(BatchingExecutor.scala:82)\n\tat scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)\n\tat scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:85)\n\tat scala.concurrent.BatchingExecutor$Batch.run(BatchingExecutor.scala:59)\n\tat scala.concurrent.Future$InternalCallbackExecutor$.unbatchedExecute(Future.scala:875)\n\tat scala.concurrent.BatchingExecutor.execute(BatchingExecutor.scala:110)\n\tat scala.concurrent.BatchingExecutor.execute$(BatchingExecutor.scala:107)\n\tat scala.concurrent.Future$InternalCallbackExecutor$.execute(Future.scala:873)\n\tat scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:72)\n\tat scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1(Promise.scala:288)\n\tat scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1$adapted(Promise.scala:288)\n\tat scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:288)\n\tat scala.concurrent.Promise.complete(Promise.scala:53)\n\tat scala.concurrent.Promise.complete$(Promise.scala:52)\n\tat scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)\n\tat scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)\n\tat scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\n\tat java.lang.Thread.run(Thread.java:748)\nCaused by: java.lang.NullPointerException\n\tat scala.meta.internal.metals.ConfiguredLanguageClient.$anonfun$metalsQuickPick$1(ConfiguredLanguageClient.scala:118)\n\tat scala.util.Success.$anonfun$map$1(Try.scala:255)\n\tat scala.util.Success.map(Try.scala:213)\n\tat scala.concurrent.Future.$anonfun$map$1(Future.scala:292)\n\tat scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)\n\t... 5 more\n', 'code': -32603, 'message': 'Internal error.'}

Reply:
Looks to me like `null` is actually sent and the server is crashing on it.

Reply:
You are right, Metals lsp trace shows that `null` was received. I was fooled by the logged `None` in the LSP console